### DISCUSSION TITLE
Closing bracket of `if`-condition on own line when condition is multiline

### DISCUSSION THREAD

**Discussion by windaishi** [PHP-CS-Fixer/PHP-CS-Fixer#7885] [Q&A] [answered]
Hello,

I am looking for a rule which helps me to ensure, that if an if-condition is multiline, that the closing bracket of the condition and the opening brace of the code block is on an own new line.

Example:
```php
if ($report->getWarehouseId() &&
    !in_array($reportId, $this->getDeletableReportIdsInWarehouses([$report->getWarehouseId()]))) {
    throw new StockValuationException();
}
```

should be fixed to:

```
if ($report->getWarehouseId() &&
    !in_array($reportId, $this->getDeletableReportIdsInWarehouses([$report->getWarehouseId()]))
) {
    throw new StockValuationException();
}
```

Thank you.

**Comment by julienfalque** [comment, marked as answer]
There's no such rule at the moment but #7813 aims to introduce it.

**Reply by windaishi** [reply]
Thank you.

**Comment by windaishi** [comment]
I found that there is a ruleset called `PSR12` for `PHP_Codesniffer` that can archive this.

**Reply by Wirone** [reply]
You can use [ECS](https://github.com/easy-coding-standard/easy-coding-standard) and take advantage of both Fixer _and_ Sniffer 😊.
